### DISCUSSION TITLE
I can't find elements by AppiumBy.accessibilityId on Appium 2

### DISCUSSION THREAD

**Discussion by LizGrillo** [appium/appium#19748] [Q&A] [answered]
I checked some discussions and Im not able to find the correct answers for me.

Preconditions: this is on iOS, native app

Previously when I had this 

    protected By continueModalButton = MobileBy.AccessibilityId("Continue"); -> this element was found

Now, with the new Appium 2: 

    protected By continueModalButton = AppiumBy.accessibilityId("Continue"); -> this element is not found.

The element Im trying to find is a button on a system modal. When I inspect the screen with the Appium inspector Im not seeing the modal. But when I inspect the screen with the Accessibility Inspector Im seeing that the button has a label. As I mentioned **before the upgrade, the scripts were capable  to find the button without any problem**. Now it's not possible. 

Can somebody help me please? If so, could you please add detailed steps since Im not a senior and I want to be able to follow the steps to solve this. 

This is how I see the button on Accessibility Inspector from Xcode: 

<img width="537" alt="Screenshot 2024-02-01 at 14 21 00" src="https://github.com/appium/appium/assets/18126612/933be243-f2c3-4acb-9473-bcd69f1dcb26">

And here's the button in the app:
<img width="297" alt="Screenshot 2024-02-01 at 14 24 43" src="https://github.com/appium/appium/assets/18126612/613b64f7-8d09-4360-9678-8e37436aaffe">

Please, I really need help 'cause I can't continue working if I cant find this button.

**Comment by KazuCocoa** [comment]
What version of xcuitest driver did you use? If you used v6, you should update your test code properly as https://github.com/appium/appium/issues/19716

**Reply by LizGrillo** [reply]
Thanks for answering @KazuCocoa!

Would you mind telling me how do I know my xcuitest driver version? 
I just installed Appium 2 with the new driver so I assume that I have the latest one.

**Reply by KazuCocoa** [reply]
If you have appium server log, it will show what version is activated. Or `appium driver list` also shows current installed one.

**Reply by KazuCocoa** [reply]
```
appium driver uninstall xcuitest
appium driver install xcuitest@5.15.1
```

let you install the 1.15.1 on your local, which may not have the change.

**Reply by LizGrillo** [reply]
- xcuitest@5.14.1 [installed (npm)] -> this is my version

**Reply by LizGrillo** [reply]
So I install v5.15.1?

**Reply by KazuCocoa** [reply, marked as answer]
Yes, 5.15.1 has a revert added in 5.14.1 which had the behavior change.
https://github.com/appium/appium-xcuitest-driver/blob/master/CHANGELOG.md#5151-2024-01-31

**Reply by LizGrillo** [reply]
Thank you, I'll try tomorrow morning!

**Reply by LizGrillo** [reply]
Fixed!! thank you so much! you saved me :)
